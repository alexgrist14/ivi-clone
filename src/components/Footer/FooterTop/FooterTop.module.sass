.top
  display: grid
  grid-template-columns: 1fr 1fr 1fr 1fr
  padding-block: 48px
  border-block: 1px solid rgba(255,255,255,.16)

  &__title
    display: block
    color: white
    margin-bottom: 8px
    font-weight: 700
    font-size: 15px
    line-height: 20px

  &__wrapper
    display: grid
    gap: 20px

  &__cert
    background-clip: text
    background-image: linear-gradient(-45deg,#c447ff,#ea003d)
    -webkit-text-fill-color: transparent
    margin-top: 8px

    & > a
      color: transparent
      font-size: 15px
      line-height: 20px
      font-weight: 500
