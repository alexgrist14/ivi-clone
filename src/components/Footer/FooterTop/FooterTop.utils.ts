export const topLeftLinks = [
  {
    text: "О компании",
    url: "https://corp.ivi.ru/?_gl=1*15hvcv0*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTI5OTQ0Ny42MC4wLjA.",
    target: "_blank",
  },
  {
    text: "Вакансии",
    url: "https://corp.ivi.ru/career/?_gl=1*2qtg9j*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.#career-vacancy-block",
    target: "_blank",
  },
  {
    text: "Программа бета-тестирования",
    url: "https://www.ivi.ru/pages/beta",
    target: "_blank",
  },
  {
    text: "Информация для партнёров",
    url: "https://www.ivi.ru/info/partners",
    target: "_blank",
  },
  {
    text: "Размещение рекламы",
    url: "https://corp.ivi.ru/advertisers/?_gl=1*uhgz7x*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.",
    target: "_blank",
  },
  {
    text: "Пользовательское соглашение",
    url: "https://www.ivi.ru/info/agreement",
    target: "_blank",
  },
  {
    text: "Политика конфиденциальности",
    url: "https://www.ivi.ru/info/confidential",
    target: "_blank",
  },
  {
    text: "Комплаенс",
    url: "https://www.ivi.ru/info/goryachaya-liniya-komplaens",
    target: "_blank",
  },
];

export const topRightLinks = [
  { text: "Мой Иви", url: "https://www.ivi.ru/", target: "_blank" },
  { text: "Что нового", url: "https://www.ivi.ru/new", target: "_blank" },
  { text: "Фильмы", url: "" },
  { text: "Сериалы", url: "" },
  { text: "Мультфильмы", url: "" },
  { text: "TV+", url: "https://www.ivi.ru/tvplus", target: "_blank" },
  {
    text: "Что посмотреть",
    url: "https://www.ivi.ru/goodmovies",
    target: "_blank",
  },
];
