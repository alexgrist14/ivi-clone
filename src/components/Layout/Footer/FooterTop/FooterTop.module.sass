@use "src/styles/vars"

.top
  display: grid
  grid-template-columns: 1fr 1fr 1fr 1fr
  gap: 15px
  padding-block: 48px
  border-block: 1px solid rgba(255,255,255,.16)

  &__title
    display: block
    color: vars.$whiteTextColor
    margin-bottom: 8px
    font-weight: 700
    font-size: 15px
    line-height: 20px

  &__wrapper
    display: grid
    gap: 20px
    white-space: pre-line

  &__cert
    background-clip: text
    background-image: linear-gradient(-45deg,#c447ff,#ea003d)
    -webkit-text-fill-color: transparent
    margin-top: 8px

    & > a
      color: transparent
      font-size: 15px
      line-height: 20px
      font-weight: 500
