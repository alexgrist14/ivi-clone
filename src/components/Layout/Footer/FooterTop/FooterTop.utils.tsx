import ILink from "/src/types/ILink";

export const topLeftLinks: Array<ILink> = [
  {
    content: "about.links.0",
    url: "https://corp.ivi.ru/?_gl=1*15hvcv0*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTI5OTQ0Ny42MC4wLjA.",
    target: "_blank",
  },
  {
    content: "about.links.1",
    url: "https://corp.ivi.ru/career/?_gl=1*2qtg9j*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.#career-vacancy-block",
    target: "_blank",
  },
  {
    content: "about.links.2",
    url: "https://www.ivi.ru/pages/beta",
    target: "_blank",
  },
  {
    content: "about.links.3",
    url: "https://www.ivi.ru/info/partners",
    target: "_blank",
  },
  {
    content: "about.links.4",
    url: "https://corp.ivi.ru/advertisers/?_gl=1*uhgz7x*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.",
    target: "_blank",
  },
  {
    content: "about.links.5",
    url: "https://www.ivi.ru/info/agreement",
    target: "_blank",
  },
  {
    content: "about.links.6",
    url: "https://www.ivi.ru/info/confidential",
    target: "_blank",
  },
  {
    content: "about.links.7",
    url: "https://www.ivi.ru/info/goryachaya-liniya-komplaens",
    target: "_blank",
  },
];

export const topRightLinks: Array<ILink> = [
  {
    content: "sections.links.0",
    url: "https://www.ivi.ru/",
    target: "_blank",
  },
  {
    content: "sections.links.1",
    url: "https://www.ivi.ru/new",
    target: "_blank",
  },
  { content: "sections.links.2", url: "/movies" },
  {
    content: "sections.links.3",
    url: "https://www.ivi.ru/series",
    target: "_blank",
  },
  {
    content: "sections.links.4",
    url: "https://www.ivi.ru/animation",
    target: "_blank",
  },
  {
    content: "sections.links.5",
    url: "https://www.ivi.ru/tvplus",
    target: "_blank",
  },
  {
    content: "sections.links.6",
    url: "https://www.ivi.ru/goodmovies",
    target: "_blank",
  },
];
