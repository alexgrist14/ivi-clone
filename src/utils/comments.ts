import { IComment } from "../types/IComment";

export const mockComment: IComment = {
  comment_id: "4d522235-7915-4122-8303-4bc79ea603a5",
  title: "About film",
  text: "My comment about the film",
  like: "36",
  film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
  createdAt: "2023-05-08T07:30:59.761Z",
  updatedAt: "2023-05-08T07:30:59.761Z",
  user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
  parent_id: null,
  user: {
    email: "admin@gmail.com",
    profile: {
      profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
      first_name: "Victor",
      last_name: "Barinov",
    },
  },
  sub_comments: [
    {
      comment_id: "ec2d35f8-b9de-4fa6-a906-c76b9f2ddaef",
      title: "About film",
      text: "My comment about the film",
      like: "36",
      film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
      createdAt: "2023-05-08T07:38:01.507Z",
      updatedAt: "2023-05-08T07:38:01.507Z",
      user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
      parent_id: "4d522235-7915-4122-8303-4bc79ea603a5",
      user: {
        email: "admin@gmail.com",
        profile: {
          profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
          first_name: "Victor",
          last_name: "Barinov",
        },
      },
      sub_comments: [
        {
          comment_id: "4c786377-f586-4580-ab6e-24cb0ccd407e",
          title: "About film",
          text: "My comment about the film",
          like: "36",
          film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
          createdAt: "2023-05-08T07:38:20.229Z",
          updatedAt: "2023-05-08T07:38:20.229Z",
          user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
          parent_id: "ec2d35f8-b9de-4fa6-a906-c76b9f2ddaef",
          user: {
            email: "admin@gmail.com",
            profile: {
              profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
              first_name: "Victor",
              last_name: "Barinov",
            },
          },
          sub_comments: [],
        },
      ],
    },
  ],
};
