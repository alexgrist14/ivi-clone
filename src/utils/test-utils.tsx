import { ReactNode } from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { reducersSetup } from "/src/store";
import { configureStore } from "@reduxjs/toolkit";
import { IProviderRender } from "../types/IRender";

export const renderWithProviders = (children: ReactNode): IProviderRender => {
  const store = configureStore({ reducer: reducersSetup });
  const component = render(<Provider store={store}>{children}</Provider>);

  return { store, component };
};
