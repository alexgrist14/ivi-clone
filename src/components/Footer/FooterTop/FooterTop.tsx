import { FC } from "react";
import styles from "./FooterTop.module.sass";
import FooterList from "../FooterList/FooterList";
import FooterButtons from "../FooterButtons/FooterButtons";
import FooterAsk from "../FooterAsk/FooterAsk";
import FooterWidget from "../FooterWidget/FooterWidget";
import { topLeftLinks, topRightLinks } from "./FooterTop.utils";
import FooterCert from "../FooterCert/FooterCert";

const FooterTop: FC = () => {
  return (
    <div className={styles.top}>
      <div>
        <span className={styles.top__title}>О нас</span>
        <nav className={styles.top__nav}>
          <FooterList items={topLeftLinks} />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>Разделы</span>
        <nav className={styles.top__nav}>
          <FooterList items={topRightLinks} />
          <FooterCert />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>Служба поддержки</span>
        <div className={styles.top__wrapper}>
          <p>
            Мы всегда готовы вам помочь.
            <br /> Наши операторы онлайн 24/7
          </p>
          <FooterButtons />
          <FooterAsk />
        </div>
      </div>
      <div>
        <FooterWidget />
      </div>
    </div>
  );
};

export default FooterTop;
