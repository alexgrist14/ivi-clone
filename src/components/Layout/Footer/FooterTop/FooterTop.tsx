import { FC } from "react";
import styles from "./FooterTop.module.sass";
import FooterList from "../FooterList/FooterList";
import FooterButtons from "../FooterButtons/FooterButtons";
import FooterAsk from "../FooterAsk/FooterAsk";
import FooterWidget from "../FooterWidget/FooterWidget";
import { topLeftLinks, topRightLinks } from "./FooterTop.utils";
import FooterCert from "../FooterCert/FooterCert";
import { useTranslation } from "next-i18next";

const FooterTop: FC = () => {
  const { t } = useTranslation("footer");

  return (
    <div className={styles.top}>
      <div>
        <span className={styles.top__title}>{t("about.title")}</span>
        <nav className={styles.top__nav}>
          <FooterList items={topLeftLinks} />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>{t("sections.title")}</span>
        <nav className={styles.top__nav}>
          <FooterList items={topRightLinks} />
          <FooterCert />
        </nav>
      </div>
      <div>
        <span className={styles.top__title}>{t("support.title")}</span>
        <div className={styles.top__wrapper}>
          <p>{t("support.text")}</p>
          <FooterButtons />
          <FooterAsk />
        </div>
      </div>
      <div>
        <FooterWidget />
      </div>
    </div>
  );
};

export default FooterTop;
