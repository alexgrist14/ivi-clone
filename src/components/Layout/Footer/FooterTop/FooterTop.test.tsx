import { renderWithProviders } from "/src/utils/test-utils";
import FooterTop from "./FooterTop";

describe("Footer top", () => {
  test("should renders without errors", () => {
    const { container } = renderWithProviders(<FooterTop />);
    expect(container).toMatchSnapshot();
  });
});
