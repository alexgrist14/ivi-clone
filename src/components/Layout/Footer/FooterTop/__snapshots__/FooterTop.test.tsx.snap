// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer top should renders without errors 1`] = `undefined`;
